src/orion/algo/dehb/dehb.py

```diff
@@ -0,0 +1,509 @@
+            Number of samples to generate
+
+        """
+        population = self.cs.sample_configuration(size=pop_size)
+        population = [
+            self.configspace_to_vector(individual) for individual in population
+        ]
+        return population
+
+    def observe(self, job_info, cost, fitness):
+        """Observe a completed job"""
+        config = job_info["config"]
+        budget = job_info["budget"]
+        parent_id = job_info["parent_id"]
+        bracket_id = job_info["bracket_id"]
+        info = dict()
+
+        #
+        for bracket in self.active_brackets:
+            if bracket.bracket_id == bracket_id:
+                # bracket job complete
+                # IMPORTANT to perform synchronous SH
+                bracket.complete_job(budget)
+
+        # carry out DE selection
+        if fitness <= self.de[budget].fitness[parent_id]:
+            self.de[budget].population[parent_id] = config
+            self.de[budget].fitness[parent_id] = fitness
+
+        # updating incumbents
+        if self.de[budget].fitness[parent_id] < self.inc_score:
+            self._update_incumbents(
+                config=self.de[budget].population[parent_id],
+                score=self.de[budget].fitness[parent_id],
+                info=info,
+            )
+
+        # book-keeping
+        self._update_trackers(
+            traj=self.inc_score,
+            runtime=cost,
+            history=(config.tolist(), float(fitness), float(cost), float(budget), info),
+        )
+
+    def seed_rng(self, seed: int) -> None:
+        """Seed the state of rngs used by DEHB"""
+        np.random.seed(seed)
+        self.cs.seed(np.random.randint(np.iinfo(np.int32).max))
+
+    @property
+    def state_dict(self) -> dict:
+        """Return a state dict that can be used to reset the state of the algorithm."""
+        state = dict(self.__dict__)
+        state["client"] = None
+        state["logger"] = None
+        for key in [
+            "active_brackets",
+            "iteration_counter",
+            "de",
+            "_max_pop_size",
+            "start",
+            "traj",
+            "runtime",
+            "history",
+        ]:
+            state[key] = getattr(self, key, None)
+
+        return deepcopy(
+            {
+                "state": state,
+                "numpy_GlobalState": np.random.get_state(),
+                "numpy_RandomState": self.cs.random.get_state(),
+            }
+        )
+
+    def set_state(self, state_dict: dict) -> None:
+        """Reset the state of DEHB"""
+        for k, v in state_dict["state"].items():
+            if hasattr(self, k):
+                setattr(self, k, v)
+            else:
+                logger.error("DEHB does not have attribute %s", k)
+
+        np.random.set_state(state_dict["numpy_GlobalState"])
+        self.cs.random.set_state(state_dict["numpy_RandomState"])
+
+
+# pylint: disable=too-many-public-methods
+class DEHB(DEHBImpl, BaseAlgorithm):
```

Comment:
Question: Why does this still need to inherit from `DEHBImpl`, even after factoring out the `DEHBImpl` portion into the `dehb` attribute?
Ideally this would only subclass `BaseAlgorithm`, and call `super().__init__(space, ...)`.

Reply:
It was a mistake indeed, I just forgot to remove it. Fixed in https://github.com/Epistimio/orion/commit/14b4f947407cc2a2c60f77b8a091a74bc0a792c1